lib/portlayer/portlayer.go

```diff
@@ -110,6 +110,20 @@ func Init(ctx context.Context, sess *session.Session) error {
 	return nil
 }
 
+func Finalize(ctx context.Context) error {
+	op := trace.NewOperation(context.Background(), "Shutdown")
+	defer trace.End(trace.Begin("", op))
+
+	storage.Finalize(op)
+	store.Finalize(ctx)
+	exec.Finalize(op)
+	network.Finalize(ctx)
+	logging.Finalize(ctx)
+	metrics.Finalize(op)
+
```

Comment:
It seems likely that we'll forget to update this if we have a new component. (However, I don't have a specific suggestion for improving this.)

Reply:
I have suggestions ;) An extension interface that has structured registration, initialization, finalization, etc. Too much for now I think.